dashboard/app/controllers/aichat_controller.rb

```diff
@@ -10,6 +10,10 @@ class AichatController < ApplicationController
   include AichatSagemakerHelper
   authorize_resource class: false
 
+  rescue_from CanCan::AccessDenied do
+    render status: :forbidden, json: {user_type: current_user&.user_type || 'signed_out'}
```

Comment:
mmm I just asked Sam for an error message for signed out users, and she shared one :). I think it might be a bit jarring/confusing to just redirect people to a sign in page with no warning when they submit a message?

Reply:
Yeah - I can see that. Just noting that for projects for signed out users, after they sign in, they are redirected back to the initial url (projects view).

Reply:
Yeah gotcha -- I think if we wanted to fully block loading AI Chat levels on having these permissions, redirecting to sign in (then back) would make total sense. But with the current approach, I like the inline error message.